integration-tests/tests/src/tests/sync/open-behavior.ts

```diff
@@ -0,0 +1,658 @@
+    const partitionValue = generatePartition();
+    const user = await this.app.logIn(Realm.Credentials.anonymous());
+
+    const config: Realm.Configuration = {
+      schema: [DogForSyncSchema],
+      sync: {
+        user,
+        partitionValue,
+        //@ts-expect-error const enum is removed at runtime, should either remove const or leave this as is.
+        _sessionStopPolicy: "immediately",
+        //@ts-expect-error const enum is removed at runtime, should either remove const or leave this as is.
+        newRealmFileBehavior: { type: "downloadBeforeOpen" },
+      },
+    };
+
+    const realm = await Realm.open(config);
+
+    expect(realm.isEmpty).to.be.true;
+
+    realm.close();
+    await user.logOut();
+  });
+  it("opening existing synced realm with downloadBeforeOpen set", async function (this: AppContext) {
+    // 1. Open empty Realm
+    // 2. Close Realm
+    // 3. Let other user upload changes to the Realm on the server.
+    // 4. Re-open empty Realm with `existingRealmFileBehavior = syncWhenOpen`
+    const partitionValue = generatePartition();
+    const returningUserCredentials = await getRegisteredEmailPassCredentials(this.app);
+    {
+      // Ensure empty realm file exists
+      const user = await this.app.logIn(returningUserCredentials);
+      const config: Realm.Configuration = {
+        schema: [DogForSyncSchema],
+        sync: {
+          user,
+          partitionValue,
+          //@ts-expect-error TYPEBUG: cannot access const enum at runtime
+          _sessionStopPolicy: "immediately",
+        },
+      };
+      const realm = await Realm.open(config);
+      realm.close();
+      await user.logOut();
+    }
+    {
+      // Update realm with different (anonymous) user
+      const user = await this.app.logIn(Realm.Credentials.anonymous());
+      const config: Realm.Configuration = {
+        schema: [DogForSyncSchema],
+        sync: {
+          user,
+          partitionValue,
+          //@ts-expect-error TYPEBUG: cannot access const enum at runtime
+          _sessionStopPolicy: "immediately",
+        },
+      };
+      const realm = await Realm.open(config);
+      realm.write(() => {
+        realm.create(DogForSyncSchema.name, { _id: new ObjectId(), name: "Milo" });
+      });
+      await realm.syncSession?.uploadAllLocalChanges();
+      realm.close();
+      await user.logOut();
+    }
+    {
+      // Check that realm contains changes made (using the same user)
+      const user = await this.app.logIn(returningUserCredentials);
+      const config: Realm.Configuration = {
+        schema: [DogForSyncSchema],
+        sync: {
+          user,
+          partitionValue,
+          //@ts-expect-error TYPEBUG: cannot access const enum at runtime
+          _sessionStopPolicy: "immediately",
+          //@ts-expect-error TYPEBUG: cannot access const enum at runtime
+          existingRealmFileBehavior: { type: "downloadBeforeOpen" },
```

Comment:
I suspect we need a TODO to fix this?

Reply:
exactly! The idea is that every place labeled with TYPEBUG is something the type system (assuming I haven't missed something) emits an error for. The idea is when bindgen is rebased on this the `ts-expect-error` will either remain the same, meaning that the error still exists, or it will get a red squiggle, which would imply that the type bug has been resolved